src/coordinator/functions/logical/common.go

```diff
@@ -66,10 +67,55 @@ func hashFunc(on bool, names ...string) func(models.Tags) uint64 {
 	return func(tags models.Tags) uint64 { return tags.IDWithExcludes(names...) }
 }
 
-// Processor is implemented by each logical transform
-type Processor interface {
-	Process(lhs block.Block, rhs block.Block) (block.Block, error)
+const (
```

Comment:
is it better to have one common place for all types or to have each type defined in their file with their implementation ? A few advantages of having them with their implementation would be that you wouldn't have one file where the mapping keeps growing. Also, you wouldn't have to change this file everytime you add a new type

Reply:
This file is only for these logical functions, so it's unlikely that it will grow, but agreed that it makes sense to keep these in their own function definition file